src/Microsoft.Azure.WebJobs.Host/Bindings/AttributeCloner.cs

```diff
@@ -161,25 +168,63 @@ internal static BindingDataResolver GetAppSettingResolver(string originalValue,
             return (newAttr, bindingData) => resolvedValue;
         }
 
+        // Run validition. This needs to be run at different stages. 
+        // In general, run as early as possible. If there are { } tokens, then we can't run until runtime.
+        // But if there are no { }, we can run statically. 
+        // If there's no [AutoResolve], [AppSettings], then we can run immediately.
+        private static void RunValidateAttributes(object value, PropertyInfo propInfo)
+        {
+            var attrs = propInfo.GetCustomAttributes<ValidationAttribute>();
```

Comment:
These attrs should be cached to save so that we don't run GetCustomAttributes every execution.

IMO the logic is way too scattered.

I think we should change the BindingDataResolver alias to a class, which caches PropInfo attributes and differentiates between dynamic / static resolvers, i.e.:

```
//instead of return (newAttr, bindingData) => resolvedValue;
return new StaticResolver(propInfo, resolvedValue);  
````

This section:
https://github.com/Azure/azure-webjobs-sdk/pull/1188/files#diff-7146d5bc5e33487d5c81156c343746abR83

would then become something like
```
var resolver = GetResolver(prop, nameResolver, bindingDataContract);
resolver.Validate();
return (Action<TAttribute, BindingData>)((attr, data) => {
  object resolved = Resolver.Resolve(attr, data); // resolve validates if dynamic resolver
  prop.SetValue(attr, resolved);
});
```